spacemesh/v2alpha1/malfeasance.proto

```diff
@@ -0,0 +1,54 @@
+syntax = "proto3";
+
+import "google/api/visibility.proto";
+
+package spacemesh.v2alpha1;
+
+message MalfeasanceProof {
+    bytes smesher = 1;
+
+    enum MalfeasanceDomain {
+        DOMAIN_UNSPECIFIED = 0; // for legacy proofs
+    }
+    MalfeasanceDomain domain = 2;
+
+    // type of the malfeasance proof, depends on domain
+    //
+    // for legacy proofs the types are
+    //   1 - Double publish of ATX
+    //   2 - Multiple ballots for a layer by same smesher
+    //   3 - Hare Equivocation (currently unused)
+    //   4 - ATX with invalid PoST proof publised
+    //   5 - ATX referencing an invalid previous ATX published
+    uint32 type = 3;
+
+    // Properties of the Malfeasance proof, different for every type of proof
+    map<string, string> properties = 4;
+}
+
+message MalfeasanceRequest {
+    repeated bytes smesher_id = 1; // list of smesher ids to fetch
+    uint64 offset = 2; // adjusts the starting point for data
+    uint64 limit = 3; // specifies max number of items to fetch
+}
```

Comment:
What's the purpose of the `offset` and `limit` fields? If N `smesher_id`s are requested, I'd expect it to return from 0 to N proofs.

Reply:
It limits the number of proofs that are returned. `smesher_id` can be empty, then all known proofs are returned, but limited to `limit` elements, starting at `offset` of all proofs. So if 1000 proofs are available you can fetch them in chunks:

- request 1: offset 0, limit 100
- request 2: offset 100, limit 100
- etc.

We do the same in other places of the API (e.g. for ATXs).

Reply:
Then the comment `// list of smesher ids to fetch` is incomplete. It should also say what you explained above. :)

Reply:
Updated